modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/AbstractDeploymentTestCase.java

```diff
@@ -1093,4 +1097,40 @@ public void initialise() throws InitialisationException {
       initialised = true;
     }
   }
+
+  protected void redeployAndVerifyPropertyInRegistry(String id, Properties deploymentProperties,
+                                                     Function<DefaultRegistry, Boolean> verifier)
+      throws IOException {
+    try {
+      deploymentService.getLock().lock();
+      redeployId(id, deploymentProperties);
+      assertConditionOnRegistry(testDeploymentListener, verifier);
```

Comment:
This can go outside the try/finally block to release the lock as soon as possible